Bake protoc in ci image

nachog00 opened the pull request:
- Bake protoc in ci image
- remove rocksdb bake as it was causing plenty of errors

nachog00 committed:
Add protobuf-compiler to Docker dependencies
- Dockerfile.ci: Added `protobuf-compiler` to the list of packages to be installed

This is required to compile protobuf files needed in the build process.

nachog00 committed:
removed baked rocksdb as it was introducing conflictswq

zancas commented:
Hmm...   I think the rocksdb bake is necessary for this image file to support building the repo when mounted into a container running on an Arch kernel.

nachog00 committed:
Add PROTOC environment variable to Dockerfile.ci
- Added `ENV PROTOC=/usr/bin/protoc` to specify the path for protoc
- Required to ensure protoc is available for builds in CI pipeline

nachog00 committed:
Comment out caching steps in CI workflow
- Commented out the 'Restore cargo dependencies' step in .github/workflows/ci.yml
- Commented out the 'Save cargo dependencies' step in .github/workflows/ci.yml

Why:
- Temporarily disable dependency caching to troubleshoot issues in the workflow.

nachog00 committed:
Add debug step to CI workflow
- Modified .github/workflows/ci.yml
- Added 'Debug env' step using hmarr/debug-action@v3
- Purpose: To aid in troubleshooting by providing environment information during CI runs

nachog00 committed:
Update Docker image tag in CI workflow
- Changed the image tag output to 'debug' in .github/workflows/ci.yml
- This change allows testing with a static tag during development/debugging

I suspect that CI is not pulling the latest version of the properly annotated tag, so im pushing it to a hardcoded 'debug' tag and see if that get's CI working.

nachog00 committed:
Update CI workflow for Docker image tag
- Changed the 'image-tag' output in 'Compute Docker Image Tag' job to use dynamic tag from 'taggen'
- Reverted 'image' in 'Build test artifacts' job to use 'debug' tag explicitly
